Redirect index to first page in sidebar, add deploy docstrings

nspevacek opened the pull request:
- Redirect index to first page in sidebar, parsed from `luma.yaml`. Refactor `_app.tsx` a bit to reduce code duplication
- Add docstrings to `luma deploy` functions. Replace `ike` reference with `luma` when saving API key to keyring

bveeramani reviewed (approved):
LGTM.

@nspevacek can you check that `npm run build` still works and that we can statically build the scaffold?

nspevacek commented:
> LGTM.
> 
> @nspevacek can you check that `npm run build` still works and that we can statically build the scaffold?

Fixed build. There was an issue in some unrelated code that I fixed here as well:
```
./components/SideNav.tsx
54:15  Error: Missing "key" prop for element in iterator  react/jsx-key
```
